### PR TITLE
Set zeros if R0 data is negative after the time lapse correction

### PR DESCRIPTION
**Pull request by SeiyaNozaki**
continued from #92 
I changed a branch name just to avoid confusion.

### CODE REVIEW

**Review by moralejo — APPROVED** [review]
Seems ok to me - and I assume you tested it in the case that raised the problem.

**Comment by SeiyaNozaki** [comment]
Yes, I tested it and it seems working well (low pedestal value as expected for the capacitors which had a problem).

File: ctapipe_io_lst/calibration.py [hunk 1]
```diff
@@ -446,6 +446,12 @@ def time_lapse_corr(self, event, tel_id):
             lst.svc.pixel_ids,
         )
 
+        # Set zeros if data is negative after time lapse correction to create a pedestal table
```
Comment (maxnoe) [inline ctapipe_io_lst/calibration.py:449]: If the timelapse correction is the code introducing this problem, why not fix it in the time lapse calibration?

**Comment by maxnoe** [comment]
This fix is not OK!

It reduces the dynamic range of the waveform from uint16 to int16 in general to fix a specific bug.

I will revert this.

**Comment by maxnoe** [comment]
What is actually the problem here? If it is only the timelapse calib resulting in an underflow, it needs to be fixed there, not by removing all values > 2¹⁶ from the waveforms.

In general these are valid values!

**Comment by maxnoe** [comment]
Also, there is now test here for this error condition!

**Comment by moralejo** [comment]
> What is actually the problem here? If it is only the timelapse calib resulting in an underflow, it needs to be fixed there, not by removing all values > 2¹⁵ from the waveforms.
> 
> In general these are valid values!

As we two discussed in Slack: currently this poses no problem because there won't be genuine values above 2e15. But it is also true that eventually EVB might produce up-scaled values to fully use the available 16 bits. And this fix would then be problematic. So fixing it in the time lapse correction is a better solution.

**Pull request merged**